pulser/_seq_drawer.py

```diff
@@ -199,6 +202,43 @@ def phase_str(phi):
         a.set_ylabel(r'$\Omega$ (rad/µs)', fontsize=14, labelpad=10)
         b.set_ylabel(r'$\delta$ (rad/µs)', fontsize=14)
 
+        if draw_phase_area:
+            top = False  # Variable to track position of box, top or center.
+            draw_phase = any(
+                seq_.type.phase != 0 for seq_ in seq._schedule[ch]
+                if isinstance(seq_.type, Pulse)
+            )
+            for pulse_num, seq_ in enumerate(seq._schedule[ch]):
+                # Select only `Pulse` objects
+                if isinstance(seq_.type, Pulse):
+                    if sampling_rate:
+                        area_val = np.sum(
+                            yaeff[seq_.ti*time_scale:seq_.tf*time_scale+1]
+                        ) / time_scale / np.pi
```

Comment:
Okay, I can understand your confusion. So, `yaeff` is not actually an array, so you can't slice it like this. Rather, it is a function to which you can give a time `t` and it will return the respective value. Also, it always expects `t` to be in µs.

On the other hand, `seq_.ti` and `seq_.tf` are always in ns, so you shouldn't multiply by `time_scale` but rather by 1e3 before you give it to `yaeff`. 

Other remarks:
- You should stop right before `seq_tf`, so you shouldn't add the `+1`. 
- After summing you should indeed adjust the value, but not by `time_scale`, by `1e3`(as is done in `Waveform.integral`)

Reply:
Thanks, I interpreted `yaeff` as a numpy array due to these lines which were already defined earlier:

```py
cs_amp = CubicSpline(solver_time, ya2)
...
yaeff = cs_amp(teff)
```

As a result, the types I get are:

```
type(cs_amp)
>> scipy.interpolate._cubic.CubicSpline
```

and

```
type(yaeff)
>> numpy.ndarray
```

Have I missed something here? Since I do get `yaeff` as an array.

Reply:
> Also, you can give cs_amp a list or array of all the times you want to evaluate over, you don't have to give it one at a time.

@HGSilveri since `teff` was already calculated, I thought the calculation would come down to just indexing on the array of times for the whole sequence. So does this solution also look fine?

Thanks!

Reply:
`yaeff` is indeed an array, which corresponds to the amplitude values at the times `teff`. When the sampling rate is 1, `teff` just goes in 1ns time steps, so that's why you seem to be able to index it like this. However, when the sampling rate is lower, this will not hold. I will make a comment on the relevant lines.